shared/analytics_tracking/manager.py

```diff
@@ -15,92 +12,8 @@ def add_tool(self, tracking_tool: BaseAnalyticsTool):
     def remove_tool(self, tracking_tool: BaseAnalyticsTool):
         self.tools.remove(tracking_tool)
 
-    def track_event(self, user_id, event_name, *, is_enterprise, event_data={}):
-        for tool in self.tools:
-            if tool.is_enabled():
-                tool.track_event(
-                    user_id,
-                    event_name,
-                    is_enterprise=is_enterprise,
-                    event_data=event_data,
-                )
-
-    def track_user(self, user_id, user_data={}, is_enterprise=False):
+    def track_event(self, event_name, *, is_enterprise, event_data={}, context=None):
```

Comment:
At this point I've had Scott mentioning to me enough times that `event_data={}` has a bug in it. The reference to the dict passed is always the same (for all function calls).

The better way of doing it is:
```python
event_data: Optional[dict] = None
if event_data is None:
    event_data = {}
```

Reply:
didn't know that, thank you!